iceoryx_dust/include/iceoryx_dust/posix_wrapper/message_queue.hpp

```diff
@@ -119,8 +135,29 @@ class MessageQueue : public DesignPattern::Creation<MessageQueue, IpcChannelErro
     /// NOLINTJUSTIFICATION used inside the wrapper so that the user does not have to use this
     ///                     construct from outside
     /// NOLINTNEXTLINE(hicpp-signed-bitwise)
-    static constexpr mode_t m_filemode{S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH};
+    static constexpr mode_t FILE_MODE{S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH};
 };
+
+class MessageQueueBuilder
+{
+    /// @brief Defines the message queue name
+    IOX_BUILDER_PARAMETER(IpcChannelName_t, name, "")
+
+    /// @brief Defines how the message queue is opened, i.e. as client or server
+    IOX_BUILDER_PARAMETER(IpcChannelSide, channelSide, IpcChannelSide::CLIENT)
+
+    /// @brief Defines the max message size of the message queue
+    IOX_BUILDER_PARAMETER(size_t, maxMsgSize, MessageQueue::MAX_MESSAGE_SIZE)
```

Comment:
Shouldn't we use here a `uint64_t` as integer type with a specified size.

Reply:
I'm dispassionate about this. I did just a 1 to 1 port without turning on my brain ;)